### PR TITLE
removed jitter in nat grads

### LINKED ISSUE #767: Jitter not handled correctly in nat grads
The nat grads param conversion code adds jitter inconsistently. This is normally fine, but if `q_sqrt` is initialized near zero it can fail

E.g.

```python
N =4
D =1 
np.random.seed(0)
X = np.random.randn(N, D)
Y = np.random.randn(N, 1)
custom_config = settings.get_settings()
custom_config.numerics.jitter_level = 1e-6

with settings.temp_settings(custom_config):

    model = SVGP(X, Y, RBF(D), Gaussian(), Z=X)

    NatGradOptimizer(gamma=1).minimize(model, var_list=[[model.q_mu, model.q_sqrt]], maxiter=1)
    print(model.compute_log_likelihood())

    model.q_mu = np.random.randn(N, 1)
    model.q_sqrt = np.eye(N)[None, :, :] * 1e-4
    NatGradOptimizer(gamma=1).minimize(model, var_list=[[model.q_mu, model.q_sqrt]], maxiter=1)
    print(model.compute_log_likelihood())
```

returns
```
-6.652198037643771
-33.57417938234626
```

They should be the same. 

Turning the jitter level down to `1e-18` returns
```
-6.652195846600538
-6.653940076332216
```

I'm using 147ce61e840ba80d3b387018c24026f32231e200, tf1.5, CPU, python3.5, ubtunu16.04

I propose to remove the jitter at https://github.com/GPflow/GPflow/blob/147ce61e840ba80d3b387018c24026f32231e200/gpflow/training/natgrad_optimizer.py#L390  which fixes this problem

### PR DESCRIPTION
**Pull request by hughsalimbeni**
Jitter removed before taking the cholesky of the covariance. Adding jitter makes the param conversions not correct, which was causing problems for situations when q_sqrt was near zero. This fixes #767 

Using the example from #767:

```python
N =4
D =1 
np.random.seed(0)
X = np.random.randn(N, D)
Y = np.random.randn(N, 1)
custom_config = settings.get_settings()
custom_config.numerics.jitter_level = 1e-6

with settings.temp_settings(custom_config):

    model = SVGP(X, Y, RBF(D), Gaussian(), Z=X)

    NatGradOptimizer(gamma=1).minimize(model, var_list=[[model.q_mu, model.q_sqrt]], maxiter=1)
    print(model.compute_log_likelihood())

    model.q_mu = np.random.randn(N, 1)
    model.q_sqrt = np.eye(N)[None, :, :] * 1e-4
    NatGradOptimizer(gamma=1).minimize(model, var_list=[[model.q_mu, model.q_sqrt]], maxiter=1)
    print(model.compute_log_likelihood())
```

This previously returned 

```
-6.652198037643771
-33.57417938234626
```

i.e. not the same.

But with this fix it returns

```
-6.6521980376374295
-6.918634344658385
```
Which is still not identical but much closer. 

Using jitter of `1e-18` we used to have 
```
-6.652195846600538
-6.653940076332216
```

now 

```
-6.652195846600538
-6.654020909627981
```
i.e. no change if the jitter was really small.

I added a new test (`test_small_q_sqrt_handeled_correctly`) which fails under the previous code but passes with the fix, using a jitter of `1e-6`

### CODE REVIEW

File: tests/test_optimizers.py [hunk 1]
```diff
@@ -269,7 +294,7 @@ def test_VGP_vs_GPR(session_tf):
     NatGradOptimizer(1.).minimize(m_vgp, [(m_vgp.q_mu, m_vgp.q_sqrt)], maxiter=1)
 
     assert_allclose(m_gpr.compute_log_likelihood(),
-                    m_vgp.compute_log_likelihood(), atol=1e-5)
+                    m_vgp.compute_log_likelihood(), atol=1e-4)
```
Comment (awav) [inline tests/test_optimizers.py]: Why did you change it?

Comment (hughsalimbeni) [reply tests/test_optimizers.py]: The new test failed at 1e-5 and I changed this one to match, but didn't to change it back. Happy to revert to 1e-5

**Comment by hughsalimbeni** [comment]
bother this now seems to be failing. I think perhaps it's the random seed (the seed is one advanced due to the new test). Perhaps the old one passing at 1e-5 was just lucky?

**Comment by awav** [comment]
@hughsalimbeni, of course they shouldn't match exactly, but I'm a bit concerned that every time we just lower precision without exploration why it is happening. I will return it back as don't think this particular thing is important and crucial.
